GOVERNANCE.md

```diff
@@ -0,0 +1,201 @@
+GitHub pull request. Collaborators review and merge (_land_) pull requests
+following the [CONTRIBUTING](CONTRIBUTING.md) guidelines.
+
+Collaborators can communicate in private with the Core Team in
+[private team discussion on GitHub][].
+
+### Collaborator activities
+
+* Helping users and novice contributors
+* Contributing code and documentation changes that improve the project
+* Reviewing and commenting on issues and pull requests
+* Merging pull requests
+* Release plugins
+
+The Core Team can remove inactive Collaborators or provide them with
+_Past Collaborators_ status. Past Collaborators may request that the Core
+Team restore them to active status.
+
+## Projects
+
+There are two kinds of projects in ESLint Community: Primary and auxiliary projects.
+
+Primary projects are maintained by Core Team in collaboration with project Collaborators
+and will be ensured to be kept up to date at a best effort pace by the Core Team.
+
+Auxiliary projects are maintained by its project Collaborators and is kept up to date at
+whatever pace those Collaborators chose. The Core Team merely fascilitates nominations
+of new Collaborators to such projects, ensuring there's a path to keep it maintained,
+without making any promises.
+
+All project npm modules should be owned (and solely owned) by the `eslint-community-bot`
+user on npm. If the npm modules are owned by an existing organisation and using its prefix,
+then the `eslint-community-bot` should become the sole owner of that organisation.
+
+Additional assets such as domain names, web sites etc should also be in control of the
+Core Team and/or the ESLint TSC.
+
+If possible, a project should always be transfered and keep using its original npm module
+name and repository. If a new module name needs to be created, it should preferably be
+`@eslint-community/<the-old-name>`.
+
+Automatic releases of the modules should be set up by the Core Team using the organization
+secret in the GitHub organization.
+
+Auxiliary projects can can be nominated to become a primary project by its Collaborators,
+by opening a [private team discussion on GitHub][].
+
+If a primary project is no longer widely depended upon, or have technical reasons to no
+longer stay a primary project, the Core Team can decide to convert it into an auxiliary
+project in discussion with its Collaborators.
+
+## Project nominations
+
+Projects that are widely depended upon within the ESLint ecosystem may be nominated to
+become ESLint-Community projects by anyone by opening a public issue on GitHub in the
+[eslint-community/governance][] repository.
+
+The project should fulfill a few criterias:
+
+The project should:
+
+* Preferably be: An ESLint plugin or formatter
+* Maybe be: A dependency used by an ESLint-Community project or the main ESLint repo
+* Not be: Primarily an ESLint config
+
+The project should also:
+
+* be widely depended upon throughout the community (eg. 3M downloads/week or similar)
```

Comment:
I see this 3M/wk example figure comes from the RFC where the majority of the mentioned projects have this level. I'm not aware of how much consideration it's had already, at RFC time, or now in being carried forward to this criterion - hence this comment.

It sounds like quite a high bar at first look, although for context ESLint itself gets 38M (so the 3M is 8%)  and the React plugin gets 18M (47%). Other stats that would be more difficult to acquire would be count or % of all plugins on npm (or https://github.com/dustinspecker/awesome-eslint) that would meet this criteria.

Any further thoughts on the figure?

As far as I know, NPM download counts and GitHub stars are the only metrics available.

Reply:
A better number / criteria would indeed be good

Eg. @aladdin-add [proposed adding `eslint-plugin-markdown`](https://github.com/eslint-community/collaborators/discussions/5) and that has 500k weekly, but I guess that was proposed more as an offspring from ESLint core.

Other possible numbers are accessible [through eg. ecosyste.ms](https://packages.ecosyste.ms/registries/npmjs.org/packages/eslint-plugin-markdown), such as number of "dependent packages" and "dependant repositories", which for eg. `eslint-plugin-markdown` is ±4k and ≈150k

At times also interesting to know which version it is that the downloads are from, eg that plugin has most of its downloads happen on the 3.x version and not yet on the latest 4.x version, which if it would be an abandoned aged project would indicate that its the 3.x version that should be adopted maybe.
![Skärmavbild 2024-04-23 kl  18 23 03](https://github.com/eslint-community/governance/assets/34457/23dcfc30-d326-4f10-b444-0ff1d9c41638)

Reply:
Some ordered searches for some visibility of the curves:

https://www.npmjs.com/search?q=eslint-plugin-&ranking=popularity
https://github.com/search?q=eslint-plugin-&type=repositories&s=stars&o=desc

I think both downloads and metrics are pretty flawed though. For popular projects I expect the share caused by public mirrors is small. Some companies run NPM mirrors and have hundreds of developers and projects hidden behind a single download count. Others are running hundreds of CI jobs daily without any caching or mirror. And how many people star the projects they use. These aren't problems for us to solve here, but to be mindful of their effect.

Reply:
What if we use number of projects that depend on the package? This might give a better idea of "importance" to the community (especially when coupled with downloads).

Reply:
I actually have been working on a tool that enables one to list dependents of a module + filter those dependencies on how old the latest release it, minimum downloads and such: https://github.com/voxpelli/list-dependents-cli

(Will be using it for some canary tests, but is similarly useful here)

On the governance text:

Maybe we just remove any hard numbers and defer it to a case by case basis?

Reply:
I think it's an useful input, but perhaps when applied to all users, a secondary one. My intuition is that there are many, many more private than public projects depending on plugins, but of course we don't have information about those.

Taking https://www.npmjs.com/package/eslint-plugin-react for example, and generously assuming that every one of the 17k dependents make 10 downloads every day of the week, that would account for 1M of the 18M weekly downloads. Limited information here so I'm clutching at straws and that might be a flawed validation of my intuition.

Reply:
Just seen the comment about removing hard numbers. I think it would be fine.

For the use case I had today at least, it would have resulted (by the word "widely") in the what I believe to be the correct outcome which is that I wouldn't have submitted my not-very-popular (3-4K/wk) small project.

There is a data gap we've come across here however that I expect is bound to have affected people making decisions (in the NPM ecosystem generally) before us and will continue to in the future.

Reply:
eslint-plugin-react has 2 million dependent repositories: https://packages.ecosyste.ms/registries/npmjs.org/packages/eslint-plugin-react

Reply:
Ah I see, dependent repositories rather than packages as I used from npmjs.org, and they have a more comprehensive view of other distribution channels for the packages number too. Presumably still not covering private projects however.

Reply:
I think we included the number because "widely used" is another vague term that could be interpreted differently by different people and just wanted to give an indication 🤷‍♂️

So I would be in favor of just removing the number completely 👍